tests/proxy.test.js

```diff
@@ -0,0 +1,178 @@
+// The following tests are run sequentially, and not in parallel
+
+import fs from 'fs'
+import path from 'path'
+import { test, expect, describe, beforeAll } from 'vitest'
+
+import { uuidv4 } from '../libs/utils'
+
+const wait = (ms) => new Promise(resolve => setTimeout(resolve, ms))
+
+const API_URL = 'http://clusterodm:3000'
+const TASK_UUID = uuidv4()
+console.log(`Testing CluserODM with task id: ${TASK_UUID}`)
+
+describe('ClusterODM basic imagery processing workflow', () => {
+  // Check the ClusterODM API is available before running tests
+  beforeAll(async () => {
+    // First, wait 3 seconds for ClusterODM --> NodeODM connection
+    console.log('Waiting 3s for ClusterODM to connect --> NodeODM')
+    await wait(3000)
+
+    const maxAttempts = 10
+    const delayMs = 1000
+
+    for (let attempt = 1; attempt <= maxAttempts; attempt++) {
+      try {
+        const response = await fetch(`${API_URL}/info`)
+        const json = await response.json()
+
+        if (json.engine === 'odm') {
+          console.log(`ClusterODM ready on attempt ${attempt}`)
+          return
+        }
+      } catch (err) {
+        console.warn(`Attempt ${attempt} to get /info failed:`, err)
+      }
+
+      await wait(delayMs)
+    }
+
+    throw new Error(`ClusterODM /info did not return engine=odm after ${maxAttempts} attempts`)
+  }, 15_000)
+
+  test('/options after node added', async () => {
+    const response = await fetch(`${API_URL}/options`)
+    const responseJson = await response.json()
+    expect(responseJson).toBeDefined()
+  })
+
+  // https://github.com/OpenDroneMap/NodeODM/blob/master/docs/index.adoc#post-tasknew
+  test('/task/new', async () => {
+    const form = new FormData()
+
+    const imagePaths = [
+      path.resolve('./tests/data/test-photo-1.jpg'),
+      path.resolve('./tests/data/test-photo-2.jpg'),
+      path.resolve('./tests/data/test-photo-3.jpg'),
+    ]
+
+    for (const filePath of imagePaths) {
+      try {
+        const blob = await fs.openAsBlob(filePath, { type: 'image/jpeg' })
+        form.append('images', blob, path.basename(filePath))
+      } catch (err) {
+        console.error(`Missing or unreadable file: ${filePath}`, err)
+      }
+    }
+
+    // Options to reduce processing time, only generate simple ortho
+    form.append('options', JSON.stringify([
```

Comment:
NodeODM has a test mode, check https://github.com/OpenDroneMap/NodeODM/blob/master/config.js#L39-L44 which might be useful for your case here.